app/grandchallenge/components/models.py

```diff
@@ -686,6 +636,67 @@ def saved_in_object_store(self):
             or not self.store_in_database
         )
 
+    @property
+    def requires_file(self):
+        return (
+            self.is_file_kind
+            or self.is_json_kind
+            and not self.store_in_database
+        )
+
+    def get_default_field(self):
```

Comment:
`get_default_field` and `get_file_mimetypes` can now be `@properties`: `default_field` and `file_mimetypes`.